config/samples/gateway-kongplugininstallation-httproute.yaml

```diff
@@ -0,0 +1,170 @@
+            - containerPort: 8080
+              name: http
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+          resources:
+            requests:
+              cpu: 10m
+---
+kind: GatewayConfiguration
+apiVersion: gateway-operator.konghq.com/v1beta1
+metadata:
+  name: kong
+  namespace: default
+spec:
+  dataPlaneOptions:
+    deployment:
+      replicas: 2
+      podTemplateSpec:
+        spec:
+          containers:
+            - name: proxy
+              # renovate: datasource=docker versioning=docker
+              image: kong/kong-gateway:3.7
+              readinessProbe:
+                initialDelaySeconds: 1
+                periodSeconds: 1
+    pluginsToInstall:
+      - name: additional-custom-plugin
+        namespace: additional
+      - name: additional-custom-plugin-2
+  controlPlaneOptions:
+    deployment:
+      podTemplateSpec:
+        spec:
+          containers:
+            - name: controller
+              # renovate: datasource=docker versioning=docker
+              image: kong/kubernetes-ingress-controller:3.2.3
+              readinessProbe:
+                initialDelaySeconds: 1
+                periodSeconds: 1
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: kong
+spec:
+  controllerName: konghq.com/gateway-operator
+  parametersRef:
+    group: gateway-operator.konghq.com
+    kind: GatewayConfiguration
+    name: kong
+    namespace: default
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: kong
+  namespace: default
+spec:
+  gatewayClassName: kong
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: httproute-echo
+  namespace: default
+  annotations:
+    konghq.com/strip-path: "true"
+    konghq.com/plugins: kong-custom-plugin,kong-custom-plugin-2
+spec:
+  parentRefs:
+    - name: kong
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /echo
+      backendRefs:
+        - name: echo
+          kind: Service
+          port: 80
+---
+apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  name: kong-custom-plugin
+plugin: additional-custom-plugin
+---
+kind: KongPluginInstallation
+apiVersion: gateway-operator.konghq.com/v1alpha1
+metadata:
+  name: additional-custom-plugin
+  namespace: additional
+spec:
+  image: northamerica-northeast1-docker.pkg.dev/k8s-team-playground/plugin-example/myheader
```

Comment:
Do those need to be named the same? If not I'd make them different in the sample so that users now which is being referred to in the `pluginsToInstall` list above. WDYT?

Reply:
Do you mean if the field `plugin: additional-custom-plugin` of `KongPlugin` has to match `name: additional-custom-plugin` field of `KongPluginInstallation`? Yet it has to, it's the same mechanism when you specify a plugin that is bundled in Kong Gateway, you have to use its name in `KongPlugin`